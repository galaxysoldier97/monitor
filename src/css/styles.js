import { typography } from '@material-ui/system';

const styles = {
  navigation: {
    paddingBottom: 15,
  },
  title: {
    fontSize: 24,
    fontWeight: typography.fontWeightLight,
    marginBottom: 20,
  },
  paper: {
    padding: 30,
  },
  clear: {
    clear: 'both',
  },
};

export default styles;
