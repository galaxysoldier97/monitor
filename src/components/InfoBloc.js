import React from 'react';
import { Grid, Paper, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/styles';
import { t } from 'mt-react-library/functions';
import PropTypes from 'prop-types';

const useStyles = makeStyles(theme => ({
  paper: {
    padding: theme.spacing(1, 4),
    backgroundColor: '#fafafa',
  },
  key: {
    color: '#5d5d5d',
    fontSize: '0.9rem',
  },
  value: {
    color: theme.palette.primary.dark,
    fontSize: '0.9rem',
  },
  paperContent: {
    overflow: 'hidden',
    display: 'block',
  },
}));

export const InfoBloc = ({label, value}) => {
  const classes = useStyles();
  return (
    <Paper className={classes.paper} elevation={0}>
      <Grid container direction="column" className={classes.paperContent}>
        <Grid item>
          <Typography className={classes.key}>{t(label)}</Typography>
        </Grid>
        <Grid item>
          <Typography noWrap variant="h5" id={label} className={classes.value}>{value === 0 ? '0' : (value || '-')}</Typography>
        </Grid>
      </Grid>
    </Paper>
  );
};

InfoBloc.propTypes = {
  label: PropTypes.string,
  value: PropTypes.string,
};
