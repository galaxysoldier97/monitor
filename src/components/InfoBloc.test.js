import React from 'react';
import renderer from 'react-test-renderer';
import { InfoBloc } from './InfoBloc';
import { createMuiTheme } from '@material-ui/core';
import { ThemeProvider } from '@material-ui/core/styles';
import { IntlProvider } from 'react-intl';

test('InfoBloc snapshot', () => {
  const theme = createMuiTheme({});
  const component = renderer.create(
    <IntlProvider key="en" locale="en" messages={{test: 'test', vtest: 'vtest'}}><ThemeProvider theme={theme}><InfoBloc label="test" value="vtest" /></ThemeProvider></IntlProvider>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
