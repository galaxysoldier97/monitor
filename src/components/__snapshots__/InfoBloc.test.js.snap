// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBloc snapshot 1`] = `
<div
  className="MuiPaper-root makeStyles-paper-1 MuiPaper-elevation0 MuiPaper-rounded"
>
  <div
    className="MuiGrid-root makeStyles-paperContent-4 MuiGrid-container MuiGrid-direction-xs-column"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <p
        className="MuiTypography-root makeStyles-key-2 MuiTypography-body1"
      >
        test
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <h5
        className="MuiTypography-root makeStyles-value-3 MuiTypography-h5 MuiTypography-noWrap"
        id="test"
      >
        vtest
      </h5>
    </div>
  </div>
</div>
`;
